Sources/Db/APIs/MySQL.php

```diff
@@ -2020,23 +2020,26 @@ protected function __construct(array $options = [])
 			$this->prefixReservedTables();
 		}
 
-		$this->get_version();
-		$this->supports_pcre = version_compare($this->version, strpos($this->version, 'MariaDB') !== false ? '10.0.5' : '8.0.4', '>=');
+		if (empty($this->connection) && !$non_fatal) {
+			$this->get_version();
+			$this->supports_pcre = version_compare($this->version, strpos($this->version, 'MariaDB') !== false ? '10.0.5' : '8.0.4', '>=');
+		}
```

Comment:
Why have these lines been wrapped in this `if` block? The test here, `(empty($this->connection) && !$non_fatal)` will never be true, because if it were we would already have bailed out higher up. As a result, the lines of code in this block will never be executed.

```suggestion
		$this->get_version();
		$this->supports_pcre = version_compare($this->version, strpos($this->version, 'MariaDB') !== false ? '10.0.5' : '8.0.4', '>=');
```

Reply:
Oh, I guess it should be !empty.  If there isn't a database connection, we can't exactly get a database version/pcre info.